### PR TITLE
feature/116834213-kc

### PR DESCRIPTION
**Pull request by kkchu791**
https://www.pivotaltracker.com/projects/1566133/stories/116834213

Set up Rails

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+gem 'rails', '>= 5.0.0.beta3', '< 5.1'
+gem 'puma'
+
+group :development, :test do
+  gem 'byebug'
+end
+
+group :development do
+  gem 'sqlite3'
```
Comment (charliemcelfresh) [inline Gemfile]: we're going to try to use redis - don't need sqlite3 here

File: bin/setup [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  puts '== Installing dependencies =='
+  system! 'gem install bundler --conservative'
+  system('bundle check') or system!('bundle install')
+
+  # puts "\n== Copying sample files =="
+  # unless File.exist?('config/database.yml')
+  #   cp 'config/database.yml.sample', 'config/database.yml'
+  # end
+
+  puts "\n== Preparing database =="
```
Comment (charliemcelfresh) [inline bin/setup]: we don't have a db -- we plan to use redis

File: bin/update [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+chdir APP_ROOT do
+  # This script is a way to update your development environment automatically.
+  # Add necessary update steps to this file.
+
+  puts '== Installing dependencies =='
+  system! 'gem install bundler --conservative'
+  system 'bundle check' or system! 'bundle install'
+
+  puts "\n== Updating database =="
+  system! 'bin/rails db:migrate'
```
Comment (charliemcelfresh) [inline bin/update]: no db !

File: config/database.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# SQLite version 3.x
```
Comment (charliemcelfresh) [inline config/database.yml]: no db

File: config/environments/production.rb [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  # Full error reports are disabled and caching is turned on.
+  config.consider_all_requests_local       = false
+  config.action_controller.perform_caching = true
+
+  # Disable serving static files from the `/public` folder by default since
+  # Apache or NGINX already handles this.
+  config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
+
+  # Compress JavaScripts and CSS.
+  config.assets.js_compressor = :uglifier
```
Comment (charliemcelfresh) [inline config/environments/production.rb]: we don't have any JS, let's remove all references to JS and assets

File: config/environments/test.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  config.action_dispatch.show_exceptions = false
+
+  # Disable request forgery protection in test environment.
+  config.action_controller.allow_forgery_protection = false
+  config.action_mailer.perform_caching = false
+
+  # Tell Action Mailer not to deliver emails to the real world.
+  # The :test delivery method accumulates sent emails in the
+  # ActionMailer::Base.deliveries array.
+  config.action_mailer.delivery_method = :test
```
Comment (charliemcelfresh) [inline config/environments/test.rb]: no mailer -- let's remove all references to mailers

File: config/initializers/active_record_belongs_to_required_by_default.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+# Be sure to restart your server when you modify this file.
+
```
Comment (charliemcelfresh) [inline config/initializers/active_record_belongs_to_required_by_default.rb]: I don't think redis has the concept of belongs to -- probably can get rid of this

File: config/initializers/cookies_serializer.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+# Be sure to restart your server when you modify this file.
```
Comment (charliemcelfresh) [inline config/initializers/cookies_serializer.rb]: This is a json api -- no cookies

File: config/puma.rb [hunk 9]
```diff
@@ -0,0 +1,47 @@
+# Puma can serve each request in a thread from an internal thread pool.
```
Comment (charliemcelfresh) [inline config/puma.rb]: we don't have a plan yet on which server to use in prod -- let's get rid of this

**Comment by rlcheng** [comment]
:+1:

**Comment by thomas-jung** [comment]
I think this looks good :+1:

**Pull request closed without merging**